lib/checks/shared/non-empty-placeholder.json

```diff
@@ -0,0 +1,14 @@
+{
+	"id": "non-empty-placeholder",
+	"evaluate": "attr-non-space-content-evaluate",
+	"options": {
+		"attribute": "placeholder"
+	},
+	"metadata": {
+		"impact": "serious",
+		"messages": {
+			"pass": "Element has a placeholder attribute",
+			"fail": "Element has no placeholder attribute or the placeholder attribute is empty"
```

Comment:
Doing this would make the generic check `attr-non-space-content` change, which in turn affects the 3 other checks that rely on it: `none-empty-alt`, `none-empty-value`, `non-empty-title` (the tests would need to be updated in each of those to handle `this.data` being called in the check).

As such, this change would be outside the scope of the PR and I vote we do this in a separate PR (the ticket you created).